src/video/video_atari.c

```diff
@@ -172,10 +222,30 @@ bool Video_Init(int screen_magnification, VideoScaleFilter filter)
 		EsetShift(TT_LOW); /* set TT 8bps video mode */
 		EgetPalette(0, 256, s_paletteBackup);	/* backup palette */
 		s_center_image_offset = 320*40;
+	} else if (s_machine_type == MCH_ST || s_machine_type == MCH_STE) {
+		/* set ST/STE 4bps video mode */
+		uint32 saddr;
+		long size = (SCREEN_WIDTH*(4+SCREEN_HEIGHT)/2) + 256;
+		saddr = (uint32) Mxalloc(size, 0);	/* allocate from ST-RAM */
+		if (saddr & 0x80000000)
+			saddr = (uint32)malloc(size);	/* use malloc() if Mxalloc is not available (TOS <= 1.04) */
+		saddr = (saddr + 255) & 0x00FFFF00;	/* 256byte alignment */
+		s_savedMode = Getrez();
+		s_savedLogBase = Logbase();
+		s_savedPhysBase = Physbase();
+		Setscreen(saddr,saddr,0);	 /* set ST-Low resolution */
```

Comment:
isn't it be possible to not allocate the screen buffer and call `Setscreen(-1, -1, 0);` to set the resolution ?

Reply:
It is, and maybe it should (it's not taking advantage of the fact it's owning the screen buffer anyway - for double buffering and whatnot)
The only current reason is to be sure we're not conflicting with VDI in a multitasking setup.

Reply:
done in 3834d3033a957ea7e2604f2da5bb70f32d47567c